libs/libc/stdlib/lib_atexit.c

```diff
@@ -0,0 +1,147 @@
+/****************************************************************************
+ * libs/libc/stdlib/lib_atexit.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <sched.h>
+#include <stdlib.h>
+
+#include "atexit.h"
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static struct atexit_list_s g_atexit_head;
```

Comment:
This is still problematic for BUILD_FLAT / BUILD_PROTECTED, as they both essentially have a flat address space and not separate process address spaces like with BUILD_KERNEL (for which this works).

Reply:
I'll implement a getter for this, the getter can obtain a re-entrant structure from the task tcb or group. Should work for flat mode too.

Reply:
Done

Reply:
g_atexit_head need move to include/nuttx/tls.h:task_info_s for flat and protected build